### PR TITLE
Automatically hide parent 'yii\bootstrap\Nav' section if all subitems are hidden

### LINKED ISSUE #117: Automatically hide parent Nav section if all subitems are hidden
Consider this example.

```php
$menuItems = [
    [
        'label' => 'Miscellaneous',
        'items' => [
            [
                'label' => 'Users',
                'url' => ['/users/index'],
                'visible' => Yii::$app->user->can('users.manage'),
            ],
            [
                'label' => 'Slides',
                'url' => ['/slides/index'],
                'visible' => Yii::$app->user->can('slides.manage'),
            ],          
            [
                'label' => 'News,
                'url' => ['/news/index'],
                'visible' => Yii::$app->user->can('news.manage'),
            ],
        ],        
    ],
];

 echo Nav::widget([
    'options' => ['class' => 'navbar-nav navbar-right'],
    'items' => $menuItems,
]);
```
If the user doesn't have permissions to manage users, slides and news, `Miscellaneous` section will be still shown, but it doesn't make sense because it doesn't have a link (click triggers visibility of subitems).

To conditionally hide it to we have to add something like this to parent section:

```php
'visible' => Yii::$app->user->can('users.manage') ||
    Yii::$app->user->can('slides.manage') ||
    Yii::$app->user->can('news.manage'),
```

This is clean violation of DRY principle. Obviously we can have a helper method for checking that, but still we need to list permissions in both places, but it's better to be done inside Nav widget itself.

So, if parent section have all subitems with `visible` property set and `visible` equals `false` in all subitems, we can just omit rendering of this whole section.

### PR DESCRIPTION
**Pull request by arogachev**
Closes #117

### CODE REVIEW

File: Nav.php [hunk 1]
```diff
@@ -202,6 +202,32 @@ public function renderItem($item)
...
+     * @return boolean
+     */
+    protected function isParentVisible($parentItem)
+    {
+        if (isset($parentItem['visible']) && !$parentItem['visible']) {
+            return false;
+        }
+
+        $items = ArrayHelper::getValue($parentItem, 'items');
+        if (empty($items)) {
```
Comment (samdark) [inline Nav.php]: Could be:

```php
$items = ArrayHelper::getValue($parentItem, 'items', []);
if ($items === []) {
```

**Comment by samdark** [comment]
@yiisoft/core-developers need additional review.

File: Nav.php [hunk 2]
```diff
@@ -202,6 +202,32 @@ public function renderItem($item)
     }
 
     /**
+     * Check if parent item is visible
+     * @param array $parentItem
+     * @return boolean
```
Comment (SilverFire) [inline Nav.php]: `@since 2.0.7`

File: Nav.php (hunk 1 above)
Comment (arogachev) [inline Nav.php]: In this case `testEmptyItems` is failing because we got `null` and `foreach` try to run on `null`.

We can add additional check for `null` but this is shorter.

File: Nav.php (hunk 2 above)
Comment (arogachev) [inline Nav.php]: Maybe `2.0.6`? This extension has no version `2.0.6.` released yet and has own version history as I see.

Comment (SilverFire) [inline Nav.php]: Yes, `2.0.6`, of course. Sorry, I've missed that we are in a separated repo

File: Nav.php [hunk 3]
```diff
@@ -202,6 +202,32 @@ public function renderItem($item)
...
+     */
+    protected function isParentVisible($parentItem)
```
Comment (klimov-paul) [inline Nav.php]: Name `isParentVisible()` is not appropriate.
Should be `isItemVisible()`, to be consistent with `isItemActive()`.

**Comment by klimov-paul** [comment]
It is better to introduce `normalizeItems()` method instead. Just as we do for regular `yii\widgets\Menu`, so those widgets will be consistent.
Besides we have #16 issue as well.

**Comment by klimov-paul** [comment]
Since now we have `BootstrapWidgetTrait`, we can actually resolve #16 and this one, extending `Nav` from `Menu`.

File: Nav.php (hunk 3 above)
Comment (arogachev) [inline Nav.php]: I think it's more appropriate because only parent items (of the first level) checked like that, and item can be considered as parent even it don't have any children.

**Comment by arogachev** [comment]
Maybe add this for now, if related issue is more difficult to implement. The amount of changes is small, so related issue can be implemented in the future,

**Comment by klimov-paul** [comment]
Adding protected method introduces new class API, which future change will cause BC break.

File: Nav.php (hunk 3 above)
Comment (klimov-paul) [inline Nav.php]: Name sounds like you are checking if item, which is parent to the given one, is chceked to be visible.

**Comment by arogachev** [comment]
@klimov-paul Could you summarize what's need to be fixed / changed in this case?

**Comment by klimov-paul** [comment]
> Could you summarize what's need to be fixed / changed in this case?

Even if do not extend `Nav` from `Menu` at this time. Thier methods interfaces should be as same as possible.
Back in `Menu` we pass `$items` through `normalizeItems()` method, which handles visibility, active check and so on. And only after that, we are rendering `$items`. Thus `Nav` should share this approach and methods API.

Actually, all we need to do is 'copy-paste' code from `Menu` to `Nav`.
